Zenventory/Services/ZFileManager.swift

```diff
@@ -7,63 +7,65 @@
 
 import SwiftUI
 
-class ZFileManager {
+internal struct ZFileManager {
 
-    static let instance = ZFileManager()
-
-    func saveImage(productImage: UIImage, name: String) {
+    internal static func saveImage(
+        productImage: UIImage,
+        name: String
+    ) throws {
         guard
             let productImgData = productImage.jpegData(compressionQuality: 1.0),
-            let productPhotoPath = getPathForImage(name: name)
+            let productPhotoPath = try getPathForImage(name: name)
         else {
-            print("Error getting Data")
-            return
+            throw ZFileManagerError.gettingData
```

Comment:
Nice one. However I'd recommend you to start custom errors everywhere :) But probably later